lib/utils/class-alerts.php

```diff
@@ -50,18 +50,24 @@ private function __construct() {
 	 *
 	 * @return array|WP_Error Response details from wp_remote_post
 	 */
-	private function send( $body ) {
-		$fallback_error = new \WP_Error( 'alerts-send-failed', 'There was an error connecting to the alerts service' );
+	private function send( array $body ) {
+		$fallback_error = new WP_Error( 'alerts-send-failed', 'There was an error connecting to the alerts service' );
 
 		$response = vip_safe_wp_remote_request( $this->service_url, $fallback_error, 3, 1, 10, [
 			'method' => 'POST',
-			'body' => json_encode( $body ),
+			'body'   => json_encode( $body ),
 		] );
 
 		if ( is_wp_error( $response ) ) {
 			return new WP_Error( 'alert-send-failed', sprintf( 'Error sending alert: %s', $response->get_error_message() ) );
 		}
 
+		// The request got a response
+		$code = $response->response->code;
```

Comment:
Two things here:
1. The result of a request is not an object but an associative array
2. Let's use Core APIs to grab the code, [see](https://developer.wordpress.org/reference/functions/wp_remote_retrieve_response_code/)